app/src/main/java/com/rockthevote/grommet/ui/views/AddressView.java

```diff
@@ -141,14 +159,37 @@ protected void onFinishInflate() {
                     sectionTitle.setText(R.string.section_label_registration_address);
             }
 
-            countyAdapter = ArrayAdapter.createFromResource(getContext(),
-                    R.array.pa_counties, android.R.layout.simple_list_item_1);
+            InputStream is = getResources().openRawResource(R.raw.pa_county_zip);
+            Writer writer = new StringWriter();
+            char[] buffer = new char[1024];
+            counties = new HashMap<>(0);
+
+            try {
+                Reader reader = new BufferedReader(new InputStreamReader(is, "UTF-8"));
+                int n;
+                while ((n = reader.read(buffer)) != -1) {
+                    writer.write(buffer, 0, n);
+                }
+
+                is.close();
+
+                counties = Counties.jsonAdapter(moshi).fromJson(writer.toString()).toHashMap();
+
+            } catch (IOException e) {
+
+            }
+
+            countyAdapter = new ArrayAdapter<>(getContext(), android.R.layout.simple_list_item_1,
+                    counties.keySet().toArray(new String[0]));
+
+            countyAdapter.sort((o1, o2) -> o1.toString().compareTo(o2.toString()));
```

Comment:
Thinking about this, we're doing IO on the main thread here. It might be good to us Rx here to mitigate that.